questionpy_server/worker/impl/subprocess.py

```diff
@@ -159,6 +193,22 @@ def _get_observation_tasks(self) -> Sequence[asyncio.Task]:
 
     async def kill(self) -> None:
         if self._proc and self._proc.returncode is None:
+            if self._exception and (
+                isinstance(self._exception, WorkerRealTimeLimitExceededError | WorkerCPUTimeLimitExceededError)
+            ):
+                # If the worker has to be killed because of a real-time or CPU-time limit, send an abort signal to the
+                # worker that will make it to dump a Python traceback (thanks to the fault handler).
+                self._proc.send_signal(signal.SIGABRT)
+
+                # We give it a short time to dump the traceback. Usually, SIGABRT should terminate the process
+                # right after Python's signal handler returns.
+                try:
+                    await asyncio.wait_for(self._proc.wait(), 0.2)
+                except TimeoutError:
+                    log.warning("Worker %s did not terminate after SIGABRT", self.name)
+                else:
+                    return
```

Comment:
Ich finde das etwas lesbarer:
```suggestion
                try:
                    await asyncio.wait_for(self._proc.wait(), 0.2)
                    return
                except TimeoutError:
                    log.warning("Worker %s did not terminate after SIGABRT", self.name)
```

Reply:
Das hatte ich ursprünglich so geschrieben und fände ich auch lesbarer, aber ruff hat mich auf https://docs.astral.sh/ruff/rules/try-consider-else/ hingewiesen.